src/bindgen/cargo/cargo.rs

```diff
@@ -247,6 +250,19 @@ impl Cargo {
             expand_default_features,
             expand_features,
             profile,
+            use_nightly,
         )
     }
+
+    pub(crate) fn path() -> Result<String, std::io::Error> {
```

Comment:
This should probably return a `Path` or a `PathBuf`. Did you need to replace the env var approach? That should have been correct.

Reply:
Yeah at least on Windows, it would appear that `CARGO` is `RUSTUP_HOME/{your_current_toolchain}/bin/cargo.exe` whereas this is `CARGO_HOME/bin/cargo.exe` - the former is tied to the toolchain and therefore doesn't support `+nightly` to easily use a sibling nightly toolchain.

```
# CARGO_HOME +nightly
λ C:\Users\bengr\.cargo\bin\cargo.exe +nightly --version
cargo 1.58.0-nightly (294967c53 2021-11-29)

# CARGO_HOME
λ C:\Users\bengr\.cargo\bin\cargo.exe --version
cargo 1.57.0 (b2e52d7ca 2021-10-21)

# RUSTUP_HOME +nightly
λ C:\Users\bengr\.rustup\toolchains\stable-x86_64-pc-windows-msvc\bin\cargo.exe +nightly --version
error: no such subcommand: `+nightly`

# RUSTUP_HOME
λ C:\Users\bengr\.rustup\toolchains\stable-x86_64-pc-windows-msvc\bin\cargo.exe --version
cargo 1.57.0 (b2e52d7ca 2021-10-21)
```


Went for string to avoid changing callers more - think it's worth changing to `PathBuf`?

Reply:
also just reverified after `rustup update` to be sure that version swap isn't related. confirmed it is not some weird `1.57` thing.